00_Base/src/services/credentials.service.ts

```diff
@@ -474,6 +477,119 @@ export class CredentialsService {
     return updatedClientInformation;
   }
 
+  async generateCredentialsTokenA(
+    credentialsRequest: CredentialsRequestDTO,
+    versionNumber: VersionNumber,
+  ): Promise<CredentialsDTO> {
+    const serverCredentialsRoleDTOs = credentialsRequest.roles;
+
+    for (const role of serverCredentialsRoleDTOs) {
+      try {
+        const storedRole =
+          await this.serverCredentialsRoleRepository.getServerCredentialsRoleByCountryCodeAndPartyId(
+            role.country_code,
+            role.party_id,
+          );
+        if (storedRole) {
+          throw new InternalServerError(
+            `ServerCredentialsRole with country_code ${role.country_code} and party_id ${role.party_id} already exists.`,
+          );
+        }
+      } catch (e) {
+        if (e instanceof NotFoundError) {
```

Comment:
Personal preference but I wish we did not have our data layer throw exception when something is not found.

I think returning `undefined` would result in cleaner code.

Reply:
Hi Lee, thanks for the review! 

+ 1. Currently this `getServerCredentialsRoleByCountryCodeAndPartyId` is used in multiple places. We might do an improvement later.